### ISSUE TITLE
Error when attempting to view Scheduled Reports (Key "name" for array with keys does not exist)

### ISSUE THREAD

**Issue by dperzanowski** [piwik/piwik#8401]
When clicking on the "Email Reports" under the Personal Administration area, I get the following error:

`Key "name" for array with keys "category, module, action, metrics, metricsDocumentation, processedMetrics, imageGraphUrl, imageGraphEvolutionUrl, uniqueId" does not exist in "@ScheduledReports/_addReport.twig" at line 150`

The installation this is happening on was updated originally from 2.0.2 to 2.12 and has been  incrementally updated 2.14.1 since the large update.

I have supplied a screenshot of the message below.

![screen shot 2015-07-22 at 1 27 09 pm](https://cloud.githubusercontent.com/assets/975970/8832178/6625f826-3075-11e5-921f-b1e27c9688ef.png)

**Comment by mattab** [comment]
Hi @dperzanowski 

in Administration > System check, do you see any warning?

Do you use any third party plugin in Administration > Plugins?

**Comment by dperzanowski** [comment]
In the system check, everything shows up with a green checkmark. As for third party plugins, we use the `cacheBuster` and `LoginLdap` plugins.

**Comment by mattab** [comment]
in the file plugins/ScheduledReports/templates/_addReport.twig at line 148, can you replace `{{ report.name|raw }}` by `{{ report.uniqueId|raw }}` and post here what reportId you see on the screen?

**Comment by dperzanowski** [comment]
Hey @mattab 

That appears to have fixed the issue of even loading the Reports page. I have added a screenshot below.

![screen shot 2015-08-10 at 2 02 28 pm](https://cloud.githubusercontent.com/assets/975970/9179156/999e3392-3f68-11e5-8f6d-406222007fe2.png)

I was also able to create a report and download it.

**Comment by mattab** [comment]
Could you also  please paste screenshot of the "New report screen" ?

**Comment by dperzanowski** [comment]
Sure thing. They are below:

![screen shot 2015-08-12 at 9 02 04 am](https://cloud.githubusercontent.com/assets/975970/9224503/e34ade5a-40d0-11e5-8b29-d0f127403e03.png)
![screen shot 2015-08-12 at 9 02 20 am](https://cloud.githubusercontent.com/assets/975970/9224504/e68fbb94-40d0-11e5-9a73-a3aa0f570237.png)

**Comment by mattab** [comment]
ok sorry, I need to try again: can you replace the new `{{ report.uniqueId|raw }}` by `{{ report.uniqueId|raw }} {% if report.name is defined %}{{report.name}}{% else %} UNDEFINED {% endif %}`

and again paste the screenshot?

**Comment by mattab** [comment]
Hi @dperzanowski can you replace the new `{{ report.uniqueId|raw }}` by `{{ report.uniqueId|raw }} {% if report.name is defined %}{{report.name}}{% else %} UNDEFINED {% endif %}`

and again paste the screenshot?

**Comment by dperzanowski** [comment]
![screen shot 2015-09-18 at 9 48 29 am](https://cloud.githubusercontent.com/assets/975970/9960758/7e2f3bb0-5dea-11e5-8b54-4ca3bbb0b6ad.png)
![screen shot 2015-09-18 at 9 48 41 am](https://cloud.githubusercontent.com/assets/975970/9960760/7f658160-5dea-11e5-8c5f-732ef7d95bc4.png)

**Comment by mattab** [comment]
Thanks for this, unfortunately @jantlwoomy I was expecting to see a `UNDEFINED` in the output but it's not there... 

Does anyone else experience this error in Piwik? `Key "name" for array with keys "category, module, action, metrics, metricsDocumentation, processedMetrics, imageGraphUrl, imageGraphEvolutionUrl, uniqueId" does not exist in "@ScheduledReports/_addReport.twig" `

**Comment by mattab** [comment]
Can't reproduce for now, please comment if you have this error

**Comment by jorgeuos** [comment]
Hi,
We are experiencing this in Matomo 3.14.1, will try to patch with suggested fix and post the output.

```
diff --git a/plugins/ScheduledReports/templates/_addReport.twig b/plugins/ScheduledReports/templates/_addReport.twig
index 11f960619f..fc96d35931 100644
--- a/plugins/ScheduledReports/templates/_addReport.twig
+++ b/plugins/ScheduledReports/templates/_addReport.twig
@@ -160,7 +160,7 @@
                                 <input type='{{ reportInputType }}' id="{{ reportType }}{{ report.uniqueId }}" report-unique-id='{{ report.uniqueId }}'
                                        name='{{ reportType }}Reports'/>
                                 <label for="{{ reportType }}{{ report.uniqueId }}">
-                                    {{ report.name|rawSafeDecoded }}
+                                    {{ report.uniqueId|rawSafeDecoded }}: {% if report.name is defined %}{{report.name|rawSafeDecoded}}{% else %} NAME UNDEFINED {% endif %}
                                     {% if report.uniqueId=='MultiSites_getAll' %}
                                         <div class="entityInlineHelp">{{ 'ScheduledReports_ReportIncludeNWebsites'|translate(countWebsites)
                                             }}</div>
```

Br, Jorge

**Comment by jorgeuos** [comment]
The error occurs when a limited user is trying to access the list of reports, and there is non.
The patch solved it, also confirmed with 4.1.1. See patch for 4.1.1 below:
```
diff --git a/plugins/ScheduledReports/templates/_addReport.twig b/plugins/ScheduledReports/templates/_addReport.twig
index b1ccb4aa6d..fbc914bee7 100644
--- a/plugins/ScheduledReports/templates/_addReport.twig
+++ b/plugins/ScheduledReports/templates/_addReport.twig
@@ -160,7 +160,7 @@
                                 <label>
                                     <input type='{{ reportInputType }}' id="{{ reportType }}{{ report.uniqueId }}" report-unique-id='{{ report.uniqueId }}'
                                            name='{{ reportType }}Reports'/>
-                                    <span>{{ report.name|rawSafeDecoded }}</span>
+                                    <span>{{ report.uniqueId|rawSafeDecoded }}: {% if report.name is defined %}{{report.name|rawSafeDecoded}}{% else %} NAME UNDEFINED {% endif %}</span>
                                     {% if report.uniqueId=='MultiSites_getAll' %}
                                         <div class="entityInlineHelp">{{ 'ScheduledReports_ReportIncludeNWebsites'|translate(countWebsites)
                                             }}</div>
```
br, Jorge
